core/src/test/java/org/apache/iceberg/hadoop/TestTableSerialization.java

```diff
@@ -51,6 +52,15 @@ public void testSerializableTable() throws IOException, ClassNotFoundException {
     table.updateSchema().addColumn("new_col", Types.IntegerType.get()).commit();
 
     TestHelpers.assertSerializedAndLoadedMetadata(table, TestHelpers.roundTripSerialize(table));
+    Table serializableTable = SerializableTable.copyOf(table);
+    TestHelpers.assertSerializedAndLoadedMetadata(
+        serializableTable, TestHelpers.KryoHelpers.roundTripSerialize(serializableTable));
+
+    table.newAppend().appendFile(FILE_A).commit();
```

Comment:
Nit: would make a new test as it seems unrelated to previous one (ie, testSerializableTableWithSnapshot)

Reply:
Moved serialization of table with snapshot to a new test.